pkg/controller/registry/resolver/resolver_test.go

```diff
@@ -409,38 +405,10 @@ func TestSolveOperators_FindLatestVersionWithNestedDependencies(t *testing.T) {
 	assert.ElementsMatch(t, expected, operators)
 }
 
-type stubCatalogSourceLister struct {
-	catsrcs   []*v1alpha1.CatalogSource
-	namespace string
-}
+type stubSourcePriorityProvider map[cache.SourceKey]int
 
-func (l *stubCatalogSourceLister) List(labels.Selector) ([]*v1alpha1.CatalogSource, error) {
-	if l.namespace == "" {
-		return l.catsrcs, nil
-	}
-	var result []*v1alpha1.CatalogSource
-	for _, cs := range l.catsrcs {
-		if cs.Namespace == l.namespace {
-			result = append(result, cs)
-		}
-	}
-	return result, nil
-}
-
-func (l *stubCatalogSourceLister) Get(name string) (*v1alpha1.CatalogSource, error) {
-	for _, cs := range l.catsrcs {
-		if cs.Name == name {
-			return cs, nil
-		}
-	}
-	return nil, errors.New("stub not found")
-}
-
-func (l *stubCatalogSourceLister) CatalogSources(namespace string) listersv1alpha1.CatalogSourceNamespaceLister {
-	return &stubCatalogSourceLister{
-		catsrcs:   l.catsrcs,
-		namespace: namespace,
-	}
+func (spp stubSourcePriorityProvider) Priority(k cache.SourceKey) int {
+	return spp[k]
```

Comment:
Do we need to null check this $somewhere to avoid a panic?

Reply:
Oh, disregard - I think this is fine as it should just return 0 when indexing on a key that doesn't exist.

/lgtm